### PR TITLE
Update to Arcade 8

### LINKED ISSUE #8969: Update to net8.0
### Context

Move from targetting, deploying and mentioning net7.0 to net8.0
The targetting update was done here: https://github.com/dotnet/msbuild/issues/8468
Other locations to update: https://github.com/dotnet/msbuild/blob/main/Directory.Build.props#L18-L26

Updating the `global.json` might result to issues in tests - that is expected to be adressed as part of this item.

cc @rainersigwald to provide more context on expected scope and timing (as we already branched off the 17.7 - fliping to net8 targetting should be fine right away - correct?)

### PR DESCRIPTION
**Pull request by rainersigwald**
Fixes #8969

- Make IDE0251 a suggestion
- Arcade 8
- Put expected and actual in the right places for some tlbimp tests
- Explicit System.IO.Compression reference

### CODE REVIEW

**Comment by JanKrivanek** [comment]
Naive question - can the eng\common\* changes associated with the arcade update be made as a maestro flow PR from arcade? (with all other fixes needed for it to go through to be added to that PR)

At least those changes comes from arcade (are not msbuild specific) and hence do not need to be reviewed in detail - is that correct?

**Comment by rainersigwald** [comment]
I generated the big change in this PR with a `darc update-dependencies` command that does locally what the darc subscriptions would create a PR for--I expected this change to break so I didn't want to change the subscription and break our 6.0 flow. You're right that `eng/common/*` belongs to Arcade and doesn't need much in the way of review.

But this is broken so don't review it at all yet :)

**Comment by JanKrivanek** [comment]
@ViktorHofer we are trying to update MSBuild TFM and Arcade version both to net8.0 and currently we are stuck at 

```
##[error].dotnet\sdk\8.0.100-preview.3.23178.7\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error PKV006: (NETCORE_ENGINEERING_TELEMETRY=Build) Target framework net7.0 is no longer supported in the latest version..
```

Should we temporarily disable the package validation, produce a successfull build (targetting net8.0) and then point baseline to that successfull build? Or is there alternative suggestion how to satisfy baseline validation check with our current baseline (targetting net7.0)?

FYI @JaynieBai, @GangWang01

**Review by JanKrivanek — APPROVED** [review]
Looks good to go to me

File: src/Build/CompatibilitySuppressions.xml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
```
Comment (ViktorHofer) [inline src/Build/CompatibilitySuppressions.xml]: please keep this comment. It is added by APICompat/PackageValidation v7 and upwards.

File: Directory.Build.props [hunk 2]
```diff
@@ -29,7 +29,7 @@
         continue to target .NET 7, so bump a 6 here to 7.
      -->
     <LatestDotNetCoreForMSBuild>$(NetCurrent)</LatestDotNetCoreForMSBuild>
-    <LatestDotNetCoreForMSBuild Condition=" '$(NetCurrent)' == 'net6.0' ">net7.0</LatestDotNetCoreForMSBuild>
+    <LatestDotNetCoreForMSBuild Condition=" '$(NetCurrent)' == 'net6.0' ">net8.0</LatestDotNetCoreForMSBuild>
```
Comment (ViktorHofer) [inline Directory.Build.props:32]: This should be deleted. NetCurrent in Arcade v8 is net8.0 already.

```suggestion
```

File: global.json [hunk 3]
```diff
@@ -3,13 +3,18 @@
     "allowPrerelease": true
   },
   "tools": {
-    "dotnet": "7.0.304",
+    "dotnet": "8.0.100-preview.3.23178.7",
```
Comment (ViktorHofer) [inline global.json:6]: Is there a reason why you don't update to the latest officially released preview (P6)?

Comment (ViktorHofer) [reply global.json:6]: Doing this incrementally is fine as well, was just wondering.

Comment (rainersigwald) [reply global.json:6]: Pretty sure it's "Rainer started this a long time ago and we haven't updated" but I also lean to doing it incrementally (given that it's almost working now).

File: Directory.Build.props (hunk 2 above)
Comment (JanKrivanek) [reply Directory.Build.props:32]: + the 2 comment lines above it should go with it

File: src/Build/CompatibilitySuppressions.xml [hunk 4]
```diff
@@ -23,17 +24,21 @@
...
   </Suppression>
   <Suppression>
     <DiagnosticId>CP0007</DiagnosticId>
     <Target>T:Microsoft.Build.Experimental.ProjectCache.ProjectCacheException</Target>
     <IsBaselineSuppression>true</IsBaselineSuppression>
   </Suppression>
-</Suppressions>
\ No newline at end of file
+  <Suppression>
+    <DiagnosticId>PKV006</DiagnosticId>
```
[the remark this answers is not in the document]
Comment (GangWang01) [reply src/Build/CompatibilitySuppressions.xml:42]: done.

**Comment by GangWang01** [comment]
After I pushed the commit that resolved review comment, the new run failed. So I reverted the commit back. The run still failed with the error below. I'm looking into it.
```
stage1\bin\bootstrap\net8.0\MSBuild\NuGet.RestoreEx.targets#L19
stage1\bin\bootstrap\net8.0\MSBuild\NuGet.RestoreEx.targets(19,5): error : (NETCORE_ENGINEERING_TELEMETRY=Restore) Static graph-based restore encountered an unhandled exception. Please file an issue at https://github.com/NuGet/Home.  The exception was:
System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
File name: 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at NuGet.Build.Tasks.Console.MSBuildStaticGraphRestore..ctor(Boolean debug)
   at NuGet.Build.Tasks.Console.Program.Main(String[] args)
```

File: global.json [hunk 5]
```diff
@@ -3,13 +3,18 @@
...
+    "runtimes": {
+        "dotnet": [
+            "8.0.0-preview.7.23363.2"
+        ]
+    },
```
Comment (ViktorHofer) [inline global.json:11]: This is an Arcade custom thing that I wouldn't use if not strictly necessary. Why do you want to bring in the runtime separately from the SDK?

Comment (GangWang01) [reply global.json:11]: It was introduced by https://github.com/dotnet/msbuild/pull/8672/commits/a54e08e73e95186b4a3135286df37cd622d4ec1e.
@rainersigwald can you help to answer?

File: src/Utilities/README.md [hunk 6]
```diff
@@ -7,6 +7,6 @@ This package contains `Microsoft.Build.Utilities.Core.dll`, which defines helper
 * [`Logger`](https://docs.microsoft.com/dotnet/api/microsoft.build.utilities.logger), a base class for custom logging functionality.
 
 ### netstandard2.0 target
-The `netstandard2.0` target of this build is configured only to output reference assemblies; at runtime MSBuild will be `net7.0` or `net472`. Please use the `net7.0`-targeted assemblies for .NET Core 6+ scenarios.
+The `netstandard2.0` target of this build is configured only to output reference assemblies; at runtime MSBuild will be `net8.0` or `net472`. Please use the `net8.0`-targeted assemblies for .NET Core 6+ scenarios.
```
[the remark this answers is not in the document]
Comment (GangWang01) [reply src/Utilities/README.md]: done, as well as src/Framework/README.md.

File: global.json (hunk 5 above)
Comment (rainersigwald) [reply global.json:11]: I think that at the time I was trying to use Arcade 8 but still target .NET 7. We shouldn't need this now--let me try removing it.

**Review by ViktorHofer — APPROVED** [review]

**Pull request merged**